### ISSUE TITLE
Switch to asymmetric signing

### ISSUE THREAD

**Issue by davidyuk** [aeternity/jwt-auth#8]
To be able to validate tokens by third-party without sharing the ability to generate them.

**Comment by mradkov** [comment]
Isn't this what the shared secret is for?

**Comment by davidyuk** [comment]
Yes, but the owner of a shared secret (in HS256 that is currently used) can validate tokens, but also he can generate them.
